### PR TITLE
Added listener to close dropdown on document click

### PR DESCRIPTION
**Pull request by shrinathprabhu**
Resolves [AR-870](https://team-1624093970686.atlassian.net/browse/AR-870).

Added an event listener on document body, close dropdown on click

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

**Review by mmjee — APPROVED** [review]
LGTM, assuming it's tested

File: src/components/lib/VDropdown/VDropdown.vue [hunk 1]
```diff
@@ -179,6 +179,18 @@ export default {
       toggle();
     }
 
+    function clickListener(e) {
```
Comment (ajithranka) [inline src/components/lib/VDropdown/VDropdown.vue:182]: We can remove the unused function parameter `e` here.

File: src/components/lib/VDropdown/VDropdown.vue [hunk 2]
```diff
@@ -151,7 +151,7 @@
 
 <script>
 import { reactive, ref } from "@vue/reactivity";
-import { watch } from "@vue/runtime-core";
+import { onBeforeUnmount, onMounted, watch } from "@vue/runtime-core";
```
Comment (ajithranka) [inline src/components/lib/VDropdown/VDropdown.vue:154]: I'm wondering if these import paths here are getting auto-completed by your editor. Vue exports its public API from its entry file, so we shouldn't need to look inside `@vue/reactivity` or `@vue/runtime-core`.

Would it be simpler to use this instead?

```suggestion
import { reactive, ref, onBeforeUnmount, onMounted, watch } from "vue";
```

File: src/components/lib/VDropdown/VDropdown.vue [hunk 3]
```diff
@@ -179,6 +179,18 @@ export default {
       toggle();
     }
 
+    function clickListener(e) {
+      isOpen.value = false;
+    }
+
+    onMounted(() => {
```
Comment (ajithranka) [inline src/components/lib/VDropdown/VDropdown.vue:186]: This logic isn't specific to the dropdown component and may be helpful in other contexts. We can potentially separate it out as a reusable directive or composition function. 

Here's [one possible implementation](https://thewebdev.info/2021/05/17/how-to-detect-clicks-outside-an-element-with-vue-js/) as a directive and one [as a composition function](https://vueuse.org/core/onclickoutside/).

We don't have to make this change at the moment — just something to keep in mind for later.

**Review by ajithranka — CHANGES_REQUESTED** [review]

File: src/components/lib/VDropdown/VDropdown.vue (hunk 2 above)
Comment (shrinathprabhu) [reply src/components/lib/VDropdown/VDropdown.vue:154]: Yeah this would be simpler. Those paths were autocompleted by the editor

**Review by ajithranka — APPROVED** [review]

**Pull request merged**